### PR TITLE
coreboot-toolchain: 4.16 -> 4.17

### PR DESCRIPTION
**Pull request by felixsinger**
###### Description of changes

Update to latest stable release.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Fails to build for me.

> error: path '/nix/store/1yisfhp4wmkkmkp3vsvb6z3x648chscc-coreboot-8da4bfe' claims to be content-addressed, but has references. This isn’t allowed

**Comment by felixsinger** [comment]
@GrahamcOfBorg build coreboot-toolchain.i386 coreboot-toolchain.arm coreboot-toolchain.aarch64 coreboot-toolchain.riscv coreboot-toolchain.ppc64 coreboot-toolchain.nds32le coreboot-toolchain.x64

File: pkgs/development/tools/misc/coreboot-toolchain/default.nix [hunk 1]
```diff
@@ -17,12 +17,12 @@ let
...
       pname = "coreboot-toolchain-${arch}";
-      version = "4.16";
+      version = "4.17";
 
       src = fetchgit {
         url = "https://review.coreboot.org/coreboot";
-        rev = version;
-        sha256 = "073n8yid3v0l9wgwnrdqrlgzaj9mnhs33a007dgr7xq3z0iw3i52";
+        rev = "8da4bfe5b573f395057fbfb5a9d99b376e25c2a4";
+        hash = "sha256-gHuqS3AvxW2fviyTvX0zMwaGeS/hY+RN4FH74ygbPfk=";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/coreboot-toolchain/default.nix:25]:
```suggestion
        hash = "sha256-lKYQxuB8QssFBDOYkOpO2BELWrigJJYKjjpzmGVoobI";
```

**Comment by felixsinger** [comment]
@GrahamcOfBorg build coreboot-toolchain.i386 coreboot-toolchain.arm coreboot-toolchain.aarch64 coreboot-toolchain.riscv coreboot-toolchain.ppc64 coreboot-toolchain.nds32le coreboot-toolchain.x64

File: pkgs/development/tools/misc/coreboot-toolchain/default.nix [hunk 2]
```diff
@@ -17,12 +17,12 @@ let
...
     stdenvNoCC.mkDerivation rec {
       pname = "coreboot-toolchain-${arch}";
-      version = "4.16";
+      version = "4.17";
 
       src = fetchgit {
         url = "https://review.coreboot.org/coreboot";
-        rev = version;
-        sha256 = "073n8yid3v0l9wgwnrdqrlgzaj9mnhs33a007dgr7xq3z0iw3i52";
+        rev = "8da4bfe5b573f395057fbfb5a9d99b376e25c2a4";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/coreboot-toolchain/default.nix:24]:
```suggestion
        rev = version;
```
https://review.coreboot.org/plugins/gitiles/coreboot/+/refs/tags/4.17

**Pull request closed without merging**